### ISSUE TITLE
Add read me files for every system design solution

### ISSUE THREAD

**Issue by IAmPramod** [npathai/system-design-with-code#16]
@npathai I would recommend adding read me files to each use case. It would help readers understand code and flow of solution,

**Comment by npathai** [comment]
@IAmPramod Yes I am working on it  If you notice I added proper instructions in README.md related to repository information and how to start the service. Just started, things will get organized slowly. Thanks for the active input!

**Comment by npathai** [comment]
Because the solution for URL service is still evolving, I will be adding hand drawn sketches, and once the solution has reached maturity, I can create a proper diagram and add it.
Thoughts @IAmPramod ?

**Comment by IAmPramod** [comment]
@npathai Yes that makes sense! I was looking at the Id generator code and if it will have flow diagram along with code, it will be much easier to understand.
Yes, I agree with you that once the solution reaches maturity, documentation will also become structured and much more informative.

Also, thanks for this wonderful initiative

**Comment by npathai** [comment]
@IAmPramod I can understand your confusion. Naming conventions are something I am actively thinking about right now. I am stuck somewhere between Anemic domain model and Domain Driven Design. That's why namings are not consistent. 
But I guess that's why this repository exists. Get first hand experience with building microservices systems and do the hard things.

References:
[Anemic Domain Model vs Domain Driven Design](https://blog.pragmatists.com/domain-driven-design-vs-anemic-model-how-do-they-differ-ffdee9371a86)

**Comment by npathai** [comment]
Maybe I can add a quick sketch in a few minutes so that it becomes easier for you to follow.
